CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+# Contributing to rust_codegen_gcc
+
+Welcome to the rust_codegen_gcc project! This guide will help you get started as a contributor. The project aims to provide a GCC codegen backend for rustc, allowing Rust compilation on platforms unsupported by LLVM and potentially improving runtime performance through GCC's optimizations.
+
+## Getting Started
+
+### Setting Up Your Development Environment
+
+1. Install the required dependencies:
+   - rustup (follow instructions on the [official website](https://rustup.rs))
+   - DejaGnu (for running libgccjit test suite)
+   - Additional packages: `flex`, `libmpfr-dev`, `libgmp-dev`, `libmpc3`, `libmpc-dev`
+
+2. Clone and configure the repository:
+   ```bash
+   git clone https://github.com/rust-lang/rust_codegen_gcc
+   cd rust_codegen_gcc
+   cp config.example.toml config.toml
+   ```
+
+3. Build the project:
+   ```bash
+   ./y.sh prepare  # downloads and patches sysroot
+   ./y.sh build --sysroot --release
+   ```
+
+### Running Tests
+
+To verify your setup:
+```bash
+# Run the full test suite
+./y.sh test --release
```

Comment:
It would be important to mention that this also runs the failing UI tests and as such, it is normal to have ~100 failures at the end.